### ISSUE TITLE
Rule Change: [class-methods-use-this] Add option to also check static methods

### ISSUE THREAD

**Issue by ilyub** [eslint/eslint#15975]
### What rule do you want to change?

class-methods-use-this

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
class A {
  // Bad
  static f() { 
    // No this
  }

  // Ok
  static g = () => { 
    // No this
  }
}

// Triggers @typescript-eslint/unbound-method rule
const f = A.f;

// Good
const g = A.g;
```

### What does the rule currently do for this code?

No warnings

### What will the rule do after it's changed?

Warn "f", but not "g"

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @ilyub, thanks for the issue!

Can you provide more details on why would using `this` be required in static methods? Static methods are a common way to expose utility functions related to the class, and they often don't need `this`. For example, in the ESLint class we have [`static outputFixes`](https://github.com/eslint/eslint/blob/f31216a90a6204ed1fd56547772376a10f5d3ebb/lib/eslint/eslint.js#L472) and [`static getErrorResults`](https://github.com/eslint/eslint/blob/f31216a90a6204ed1fd56547772376a10f5d3ebb/lib/eslint/eslint.js#L497) that don't use `this`. What should the user do to fix the lint problem?

**Comment by mdjermanovic** [comment]
> that works just the same, but also will not trigger "@typescript-eslint/unbound-method" rule.

Thanks for clarifying! I'm not in favor of this change because I don't think this should be a responsibility of the `class-methods-use-this` rule, and I don't think that using arrow functions instead of static methods that don't use `this` generally makes the code better. Given that there's already another thumb down from a team member, it's very unlikely that we will accept this proposal so I'm closing this issue.
